tests/bluetooth/bsim_bt/bsim_test_eatt/src/main_collision.c

```diff
@@ -0,0 +1,215 @@
+#include <sys/printk.h>
+#include <sys/util.h>
+#include <sys/byteorder.h>
+
+#include <bluetooth/bluetooth.h>
+#include <bluetooth/conn.h>
+#include <bluetooth/att.h>
+
+#include "bs_types.h"
+#include "bs_tracing.h"
+#include "bstests.h"
+
+extern enum bst_result_t bst_result;
+
+#define FAIL(...)                                                                                  \
+	do {                                                                                       \
+		bst_result = Failed;                                                               \
+		bs_trace_error_time_line(__VA_ARGS__);                                             \
+	} while (0)
+
+#define PASS(...)                                                                                  \
+	do {                                                                                       \
+		bst_result = Passed;                                                               \
+		bs_trace_info_time(1, __VA_ARGS__);                                                \
+	} while (0)
+
+static struct bt_conn *default_conn;
+
+static const struct bt_data ad[] = {
+	BT_DATA_BYTES(BT_DATA_FLAGS, (BT_LE_AD_GENERAL | BT_LE_AD_NO_BREDR)),
+};
+
+static bool volatile is_connected;
+
+static void connected(struct bt_conn *conn, uint8_t conn_err)
+{
+	char addr[BT_ADDR_LE_STR_LEN];
+
+	bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
+
+	if (conn_err) {
+		bt_conn_unref(default_conn);
+		default_conn = NULL;
+		FAIL("Failed to connect to %s (%u)\n", addr, conn_err);
+	}
+
+	default_conn = conn;
+	printk("Connected: %s\n", addr);
+	is_connected = true;
+}
+
+static void disconnected(struct bt_conn *conn, uint8_t reason)
+{
+	char addr[BT_ADDR_LE_STR_LEN];
+
+	bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
+
+	printk("Disconnected: %s (reason 0x%02x)\n", addr, reason);
+
+	if (default_conn != conn) {
+		FAIL("Conn mismatch disconnect %s %s)\n", default_conn, conn);
+	}
+
+	default_conn = NULL;
+	is_connected = false;
+}
+
+BT_CONN_CB_DEFINE(conn_callbacks) = {
+	.connected = connected,
+	.disconnected = disconnected,
+};
+
+static void test_peripheral_main(void)
+{
+	int err;
+	uint16_t num_enhanced;
+
+	err = bt_enable(NULL);
+	if (err) {
+		FAIL("Can't enable Bluetooth (err %d)\n", err);
+	}
+
+	err = bt_le_adv_start(BT_LE_ADV_CONN_NAME, ad, ARRAY_SIZE(ad), NULL, 0);
+	if (err) {
+		FAIL("Advertising failed to start (err %d)\n", err);
+	}
+
+	while (!is_connected) {
+		k_sleep(K_MSEC(100));
+	}
+
+	err = bt_eatt_connect(default_conn, CONFIG_BT_EATT_MAX);
+	if (err) {
+		FAIL("Sending credit based connection request failed (err %d)\n", err);
+	}
+
+	/* TODO: Use a flag set vy callback from EATT connection instead of sleeping */
+	/* Wait for EATT channels to be connected */
+	k_sleep(K_MSEC(10000));
+	num_enhanced = bt_att_num_eatt(default_conn);
+	printk("%d enhanced bearers connected\n", num_enhanced);
+	if (num_enhanced != CONFIG_BT_EATT_MAX) {
```

Comment:
Perhaps a bit CPU expensive, but with `bt_att_num_eatt` we can actually check that value in a loop instead of waiting 10 seconds. Something to consider :)

Reply:
Because I use the test in #42359 as well, the test can not be done when `CONFIG_BT_EATT_MAX` channels have been created. Since this will be rewritten to use the callback anyway, I think I'll keep it as is for now.